Formula/nginx-unit.rb

```diff
@@ -0,0 +1,267 @@
+      "--control=unix:#{var/"run/unit.control.sock"}",
+      "--cc-opt=-I#{pcre.opt_include} -I#{openssl.opt_include}",
+      "--ld-opt=-L#{pcre.opt_lib} -L#{openssl.opt_lib}"
+
+    system "make", "install"
+
+    # Beyond here we want to install modules into the keg, so we'll move them
+    # into the prefix ourselves
+    libunit = lib/"unit"
+    libunit.mkpath
+
+    # There is no system Go
+    #system "./configure", "go"
+    #system "make", "go"
+    #libunit.install "build/go.unit.so"
+
+    system "./configure", "perl"
+    system "make", "perl"
+    libunit.install "build/perl.unit.so"
+
+    # We can't build against the system PHP, it doesn't have the right SAPI
+    #system "./configure", "php"
+    #system "make", "php"
+    #libunit.install "build/php.unit.so"
+
+    system "./configure", "python"
+    system "make", "python"
+    libunit.install "build/python.unit.so"
+
+    system "./configure", "ruby"
+    system "make", "ruby"
+    libunit.install "build/ruby.unit.so"
+  end
+
+  def caveats
+    <<~EOS
+      Once running, you can control unit using the control socket, for example using curl:
+        curl --unix-socket #{var}/run/unit.control.sock http://localhost/
+
+      Modules are available for system perl, python and ruby. The system php doesn't support nginx-unit. You can install additional modules into:
+        #{HOMEBREW_PREFIX/"lib/unit"}
+    EOS
+  end
+
+  plist_options :manual => "unitd"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <false/>
+        <key>ProgramArguments</key>
+        <array>
+            <string>#{opt_bin}/unitd</string>
+            <string>--no-daemon</string>
+        </array>
+        <key>WorkingDirectory</key>
+        <string>#{HOMEBREW_PREFIX}</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/unitd", "--version"
+
+    require "json"
+    require "net/http"
+    require "socket"
+    require "timeout"
+
+    logfile = var/"log/unit.log"
+    logfile.unlink if logfile.exist?
+
+    pidfile = var/"run/unit.pid"
+    pidfile.unlink if pidfile.exist?
+
+    begin
+      # We need to install rack for the ruby module test to work, so set a
+      # temporary gems path before booting unitd. (unitd boots and functions
+      # fine without, but fails to start an app until rack is installed.)
+      rubydir = testpath/"ruby"
+      rubygems = rubydir/"gems"
+      rubygems.mkpath
+      ENV["GEM_PATH"] = rubygems.to_s
+      system "gem", "install", "rack", "--install-dir=#{rubygems}", "--verbose"
+
+      pid = spawn "#{bin}/unitd", "--no-daemon"
+
+      timeout(5) { sleep 0.1 until pidfile.exist? }
```

Comment:
This feels a bit... hacky. How unlikely is it the pidfile wouldn't exist in this scenario?

Reply:
It might take a moment for unitd to boot and write the pidfile... depending on the speed of the system performing the test. I didn't want to leave a race condition that might become flakey. I did have a `sleep 2` like some other formulae but I got tired of waiting while re-running the tests. I can remove, or revert to `sleep 2`, or something, if you prefer?